### PR TITLE
Add and improve tests in the utils/ folder

### PR DESCRIPTION
**Pull request by trevorgerhardt**
Added and updated tests in the utils folder. An effort to get better coverage.

### CODE REVIEW

File: lib/utils/__tests__/modification.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import toStartCase from 'lodash/startCase'
+
+import {MODIFICATION_TYPES} from '../../constants'
+import {create, formatForServer, isValid, validateSegments} from '../modification'
+import {mockModification} from '../mock-data'
+
+describe('Utils > Modification', () => {
+  describe('create', () => {
+    it('should create a modification of each type', () => {
+      MODIFICATION_TYPES.forEach(t => {
```
Comment (evansiroky) [inline lib/utils/__tests__/modification.js]: Let's make this into individual test cases for each modification type by creating an `it` test case within the forEach loop of each modification type.

Ex:  

```javascript
MODIFICATION_TYPES.forEach(t => {
  it(`should create a modification of type ${t}`, () => {
```

File: lib/utils/__tests__/get-line-string.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const end = [-70.01, 40]
+    expect(await getLineString(start, end, {followRoad: false})).toEqual({
+      coordinates: [[-70, 40], [-70.01, 40]],
+      type: 'LineString'
+    })
+
+    nock('https://graphhopper.com')
+      .get('/api/1/route')
+      .query(true)
+      .reply(200, mockResponse)
```
[the remark this answers is not in the document]
Comment (evansiroky) [reply lib/utils/__tests__/get-line-string.js]: yep.  :)

File: lib/components/modification/reroute.js [hunk 3]
```diff
@@ -82,136 +64,143 @@ export default class RerouteComponent extends Component<void, Props, void> {
...
-    update(modification)
-    this.props.setMapState({
-      allowExtend: modification.toStop === null || modification.fromStop === null,
-      extendFromEnd: modification.toStop === null,
-      spacing: 0,
-      state: MAP_STATE_TRANSIT_EDITOR
+  _clearFromStop = () => {
+    const p = this.props
+    p.update({
+      fromStop: null,
```
[the remark this answers is not in the document]
Comment (trevorgerhardt) [reply lib/components/modification/reroute.js]: Correct, it updates the modification.

File: lib/components/modification/reroute.js [hunk 4]
```diff
@@ -82,136 +64,143 @@ export default class RerouteComponent extends Component<void, Props, void> {
...
+                  onClick={this._selectFromStop}
+                  style='info'
+                  title='Select from stop'
+                >
                   <Icon type='crosshairs' /> {message('common.select')}
                 </Button>
-                <Button block onClick={this._clearFromStop} style='danger'>
+                <Button
+                  block
+                  disabled={p.modification.fromStop == null}
```
[the remark this answers is not in the document]
Comment (trevorgerhardt) [reply lib/components/modification/reroute.js]: Ya, just made operations clearer.

File: lib/utils/__tests__/get-line-string.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (trevorgerhardt) [reply lib/utils/__tests__/get-line-string.js]: Correct, so we still need another level of tests.

File: lib/utils/__tests__/get-route-polyline.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      .reply(200, (uri, body) => {
+        expect(uri.indexOf(lonlat.toLatFirstString(start)) > -1).toBeTruthy()
+        expect(uri.indexOf(lonlat.toLatFirstString(end)) > -1).toBeTruthy()
+        return mockResponse
+      })
+
+    getRoutePolyline(start, end)
+      .then(polyline => {
+        expect(polyline).toHaveLength(20)
+        expect(polyline[0]).toEqual([-77.01141, 38.93132])
```
[the remark this answers is not in the document]
Comment (trevorgerhardt) [reply lib/utils/__tests__/get-route-polyline.js]: That would probably get into the realm of testing `decodePolyline` from `@mapbox/polyline`.  We're mainly checking whether or not our code ran properly while using that library.

**Pull request merged**